### PR TITLE
chore: update jsoncons dependency

### PR DESCRIPTION
**Pull request by iko1**
The next version of jsoncons includes a few bug fixes and performance enhancements
Moreover, it contains the fix that allows the jsoncons to use mimalloc allocator instead of the standard allocator.

### CODE REVIEW

**Comment by romange** [comment]
@iko1 seems like the new version does not compile in Dragonfly.

**Comment by iko1** [comment]
> @iko1 seems like the new version does not compile in Dragonfly.

A fix will be sent shortly

File: src/core/json_test.cc
Comment (romange) [inline src/core/json_test.cc:108]: you can use `std::pmr::vector`

File: src/core/json_test.cc
Comment (romange) [inline src/core/json_test.cc:117]: same

**Comment by iko1** [comment]
Thanks to @royjacobson , he opened an issue regarding the current compilation error in the jsoncons repo
In order to complete this PR, I suggest waiting until the maintainer of the jsoncons project will release a fix that lets up use pmr::polymorphic_allocator **or** use an alternative allocator that can be inspired from https://github.com/danielaparker/jsoncons/blob/master/test/common/FreeListAllocator.hpp but with use of mimalloc.

**Comment by royjacobson** [comment]
@iko1 I think the allocator fixes didn't got into the 0.170 release branch.

**Comment by iko1** [comment]
The changes are currently on the version 0.170.2 of jsoncons.
I'm working on fixing this PR

**Comment by iko1** [comment]
> @iko1 I think the allocator fixes didn't got into the 0.170 release branch.

The allocator fixes got into the jsoncons 0.171.0 release

**Comment by royjacobson** [comment]
@iko1 thanks, looks like the new version works correctly! We're going to have a new release soon, so to minimize disruption I'm going to merge this right afterwards.

**Review by romange — APPROVED** [review]

**Pull request merged**
